hibernate-core/src/main/java/org/hibernate/action/internal/EntityUpdateAction.java

```diff
@@ -267,6 +266,89 @@ else if ( session.getCacheMode().isPutEnabled() ) {
+
+		if (instance instanceof PersistentAttributeInterceptable) {
+			PersistentAttributeInterceptable interceptable = ((PersistentAttributeInterceptable) instance );
+			PersistentAttributeInterceptor interceptor = interceptable.$$_hibernate_getInterceptor();
+			final BytecodeEnhancementMetadata enhancementMetadata = entityMetamodel.getBytecodeEnhancementMetadata();
+			if (interceptor == null) {
+				persister.getEntityMetamodel().getBytecodeEnhancementMetadata().injectInterceptor(instance, entry.getEntityKey(), session);
+				interceptor = enhancementMetadata.extractLazyInterceptor(instance);
+
+				Type[] types = persister.getPropertyTypes();
+				final String[] propertyNames = persister.getPropertyNames();
+				final CascadeStyle[] cascadeStyles = persister.getPropertyCascadeStyles();
+				for ( int i = 0; i < types.length; i++) {
+					final String propertyName = propertyNames[i];
+					CascadeStyle cascadeStyle = cascadeStyles[i];
+					if(cascadeStyle != CascadeStyles.NONE) {
+						interceptor.attributeInitialized(propertyName);
+					}
+				}
+			}
+		}
+		else {
+			if (persister.hasProxy()) {
+				Type[] types = persister.getPropertyTypes();
+				final CascadeStyle[] cascadeStyles = persister.getPropertyCascadeStyles();
+				PersistenceContext persistenceContext = session.getPersistenceContextInternal();
+				for ( int i = 0; i < types.length; i++) {
+					if(!types[i].isAssociationType()) {
+						continue;
+					}
+
+					final AssociationType type = (AssociationType) types[i];
+					if(type instanceof EntityType) {
+						EntityType entityType = (EntityType) type;
+						if (entityType.isEager(null)) {
+							continue;
+						}
+					}
+
+					CascadeStyle cascadeStyle = cascadeStyles[i];
+					if(cascadeStyle != CascadeStyles.NONE) {
+						continue;
+					}
+					Object propertyValue = persister.getEntityTuplizer().getGetter(i).get(instance);
+
+					if(types[i].isCollectionType()) {
+						continue;
+					}
+
+					if(propertyValue == null) {
+						continue;
+					}
+					else {
+						if(propertyValue instanceof HibernateProxy) {
+							continue;
+						}
+						EntityPersister propertyPersister = session.getEntityPersister( null, propertyValue );
+						if(propertyPersister.getEntityTuplizer().getProxyFactory() == null) {
+							continue;
+						}
+						if(!propertyPersister.canExtractIdOutOfEntity()) {
+							continue;
+						}
+						Serializable entityId = propertyPersister.getEntityTuplizer().getIdentifier(propertyValue, session);
+						Object proxy = null;
+						try {
+							proxy = propertyPersister.createProxy(entityId, session);
+							if(propertyValue != null) {
+								Hibernate.initialize(proxy);
+							}
+						}
+						catch (Exception e) {
+							continue;
+						}
+						final EntityKey keyToLoad = session.generateEntityKey( entityId, propertyPersister );
+						persistenceContext.addProxy( keyToLoad, proxy );
+						persister.getEntityTuplizer().setPropertyValue(instance, i, proxy);
```

Comment:
I think the changes to CollectionType and AbstractEntityTuplizer ended up taking care of some of the reasons this code was initially helpful.  I was able to remove this code without any impact to our test cases.

I was also trying to keep the changes between EntityUpdateAction and EntityInsertAction consistent with each other, defensively in case there is regression, but as of today this code wasn't actually executing anymore so we might as well get rid of it.